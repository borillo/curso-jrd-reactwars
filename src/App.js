import React from 'react';
import { BrowserRouter, Switch, Route } from 'react-router-dom';

import { 
  Navbar,
  Nav,
  NavItem,
  MenuItem
} from 'react-bootstrap';

import FilmsPanel from './films';
import AboutPanel from './about';

export default () => {
    return (
      <BrowserRouter>
        <div className="App">
          <Navbar>
            <Navbar.Header>
              <Navbar.Brand>
                <a href="/">ReactWars</a>
              </Navbar.Brand>
            </Navbar.Header>
            <Nav>
              <NavItem eventKey={1} href="/films">Films</NavItem>
              <MenuItem eventKey={2} href="/about">About</MenuItem>
            </Nav>
          </Navbar>

          <Switch>
            <Route exact path="/" component={FilmsPanel}/>
            <Route path="/films" component={FilmsPanel}/>
            <Route path="/about" component={AboutPanel}/>
          </Switch>
        </div>
      </BrowserRouter>
    );
};
