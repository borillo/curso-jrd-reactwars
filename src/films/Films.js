import React, { Component } from 'react';

import FilmsRepository from './FilmsRepository';

export default class Films extends Component {
  constructor(props) {
    super(props);

    this.state = {
      films: []
    }
  }

  componentWillMount() {
    FilmsRepository.retrieveFilms()
      .then((response) => {
        this.setState({
          films: response.results
        });
      });
  }

  render() {
    return (
      <div>
        {
          this.state.films.map((film) => {
            return (
              <div className="film" key={film.episode_id}>
                {film.title}
              </div>
            );
          })
        }
      </div>
    );
  }
}