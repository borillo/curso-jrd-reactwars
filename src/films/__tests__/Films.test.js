jest.mock('../Repository');

import React from 'react';
import { mount } from 'enzyme';

import App from '../../App';
import Films from '../Films';

const FILM_TITLES = [
  'The Phantom Menace', 
  'Attack of the Clones', 
  'Revenge of the Sith', 
  'A New Hope', 
  'The Empire Strikes Back', 
  'Return of the Jedi', 
  'The Force Awakens'
];

const FILM_EPISODES = FILM_TITLES.map((title, index) => index+1);
const NUMBER_OF_FILMS = FILM_TITLES.length;

describe('Films', () => {
  let wrapper;
  let films;

  beforeEach(() => {
    wrapper = mount(<App />);
    films = new Films(wrapper);
  });

  it('should be listed', async () => {
    expect(films.obtainFilms()).toHaveLength(NUMBER_OF_FILMS);   
  });

  it('should show name and episode number', async () => {
    const filmsTitles = films.obtainFilmsTitles();
    expect(filmsTitles).toEqual(FILM_TITLES);

    const filmsEpisodes = films.obtainFilmsEpisodes();
    expect(filmsEpisodes).toEqual(FILM_EPISODES);
  });

  it('should be ordered by episode number', async () => {
    films.obtainFilmsEpisodes().forEach((episode, index) => {
      expect(episode).toEqual(index + 1);
    });
  });
});
