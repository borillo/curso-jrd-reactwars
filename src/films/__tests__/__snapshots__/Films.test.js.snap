// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Films should match snapshot 1`] = `
<Component>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <div
        className="App"
      >
        <Switch>
          <Route
            component={[Function]}
            computedMatch={
              Object {
                "isExact": true,
                "params": Object {},
                "path": "/",
                "url": "/",
              }
            }
            exact={true}
            location={
              Object {
                "hash": "",
                "pathname": "/",
                "search": "",
                "state": undefined,
              }
            }
            path="/"
          >
            <FilmsPanel
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
              location={
                Object {
                  "hash": "",
                  "pathname": "/",
                  "search": "",
                  "state": undefined,
                }
              }
              match={
                Object {
                  "isExact": true,
                  "params": Object {},
                  "path": "/",
                  "url": "/",
                }
              }
              staticContext={undefined}
            >
              <div />
            </FilmsPanel>
          </Route>
        </Switch>
      </div>
    </Router>
  </BrowserRouter>
</Component>
`;
